Frameworks/CoreText/DWriteWrapper.mm

```diff
@@ -732,45 +762,16 @@ static void __InitDWriteFontPropertiesFromName(
 
-        // Possible optimization here that can be done by using a dictionary to functions,
+        // Possible optimization here that can be done by using a dictionary,
         // but seems excessive given that font names generally don't have more than three modifiers
-        if (CFEqual(propertyToken, CFSTR("THIN"))) {
-            *pPropertyContext->weight = DWRITE_FONT_WEIGHT_THIN;
-        } else if (CFEqual(propertyToken, CFSTR("EXTRALIGHT"))) {
-            *pPropertyContext->weight = DWRITE_FONT_WEIGHT_EXTRA_LIGHT;
-        } else if (CFEqual(propertyToken, CFSTR("ULTRALIGHT"))) {
-            *pPropertyContext->weight = DWRITE_FONT_WEIGHT_ULTRA_LIGHT;
-        } else if (CFEqual(propertyToken, CFSTR("LIGHT"))) {
-            *pPropertyContext->weight = DWRITE_FONT_WEIGHT_LIGHT;
-        } else if (CFEqual(propertyToken, CFSTR("SEMILIGHT"))) {
-            *pPropertyContext->weight = DWRITE_FONT_WEIGHT_SEMI_LIGHT;
-            // skip since this is the default
-            // } else if (CFEqual(propertyToken, CFSTR("NORMAL"))) {
-            //     *pPropertyContext->weight = DWRITE_FONT_WEIGHT_NORMAL;
-        } else if (CFEqual(propertyToken, CFSTR("REGULAR"))) {
-            *pPropertyContext->weight = DWRITE_FONT_WEIGHT_REGULAR;
-        } else if (CFEqual(propertyToken, CFSTR("MEDIUM"))) {
-            *pPropertyContext->weight = DWRITE_FONT_WEIGHT_MEDIUM;
-        } else if (CFEqual(propertyToken, CFSTR("DEMIBOLD"))) {
-            *pPropertyContext->weight = DWRITE_FONT_WEIGHT_DEMI_BOLD;
-        } else if (CFEqual(propertyToken, CFSTR("SEMIBOLD"))) {
-            *pPropertyContext->weight = DWRITE_FONT_WEIGHT_SEMI_BOLD;
-        } else if (CFEqual(propertyToken, CFSTR("BOLD"))) {
-            *pPropertyContext->weight = DWRITE_FONT_WEIGHT_BOLD;
-        } else if (CFEqual(propertyToken, CFSTR("EXTRABOLD"))) {
-            *pPropertyContext->weight = DWRITE_FONT_WEIGHT_EXTRA_BOLD;
-        } else if (CFEqual(propertyToken, CFSTR("ULTRABOLD"))) {
-            *pPropertyContext->weight = DWRITE_FONT_WEIGHT_ULTRA_BOLD;
-        } else if (CFEqual(propertyToken, CFSTR("BLACK"))) {
-            *pPropertyContext->weight = DWRITE_FONT_WEIGHT_BLACK;
-        } else if (CFEqual(propertyToken, CFSTR("HEAVY"))) {
-            *pPropertyContext->weight = DWRITE_FONT_WEIGHT_HEAVY;
-        } else if (CFEqual(propertyToken, CFSTR("EXTRABLACK"))) {
-            *pPropertyContext->weight = DWRITE_FONT_WEIGHT_EXTRA_BLACK;
-        } else if (CFEqual(propertyToken, CFSTR("ULTRABLACK"))) {
-            *pPropertyContext->weight = DWRITE_FONT_WEIGHT_ULTRA_BLACK;
-
-        } else if (CFEqual(propertyToken, CFSTR("UNDEFINED"))) {
+        for (const auto& weightMapping : c_weightMap) {
+            if (CFEqual(propertyToken, weightMapping.stringValue)) {
+                *pPropertyContext->weight = weightMapping.dwriteValue;
+                break;
+            }
+        }
+
+        if (CFEqual(propertyToken, CFSTR("UNDEFINED"))) {
```

Comment:
for small N's (small == 5) it would be true.  Are you saying I can't say one thing and say entirely another thing later :)?

---
In reply to: [80598679](https://github.com/Microsoft/WinObjC/pull/1039#discussion_r80598679) [](ancestors = 80598679,80598528)